internal/httpserver/address.go

```diff
@@ -0,0 +1,7 @@
+package httpserver
+
+// GetAddress obtains the address the HTTP server is listening on.
+func (s *Server) GetAddress() (address string) {
+	<-s.addressSet
+	return s.address
+}
```

Comment:
is this used anywhere?

Reply:
It might be used in one of the pprof package test to find what port the server is listening on. I'll double check, please leave this unresolved.

Reply:
Yes it's used in a test in the `internal/pprof` package ([here](https://github.com/ChainSafe/gossamer/blob/8985412277b6c700b847748768f2a7ce76c872a9/internal/pprof/server_test.go#L42)), so it has to stay. It might be useful for future http servers to run tests on a randomly assigned port (aka run in parallel).